roles/create-ai-environment/tasks/main.yaml

```diff
@@ -0,0 +1,274 @@
+      apt install -y python-pip
+      pip install pygithub
+
+      python '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/tools/update_env_info.py' \
+      --job {{ job_name }} --patch "{{ zuul.patchset }}" --user '{{ moo_github_account.user_name }}' \
+      --password '{{ moo_github_account.user_password }}' \
+      --url "http://logs.openlabtesting.org/logs/{{ zuul.change[-2:] }}/{{ zuul.change }}/{{ zuul.patchset }}/check/{{ zuul.job }}/{{ zuul.build[:7] }}/test_results/" \
+      --ip '{{ nodepool.public_ipv4 }}' --keeping '{{ env_keeping | lower }}'
+    executable: /bin/bash
+
+- name: Generate github raw file link for MOO
+  shell:
+    cmd: |
+      curl -L "https://github.com/moo-ai/moo-ai.github.io/raw/master/env_info/task_list.json"
+      curl -L "https://github.com/moo-ai/moo-ai.github.io/raw/master/env_info/task_list_sleep.json"
+    executable: /bin/bash
+
+- name: Get User Metadata Zip or Url
+  shell:
+    cmd: |
+      set -xe
+
+      apt install -y unzip
+      pip install pyyaml
+
+      mkdir -p "{{ result_folder }}"
+
+      cd '{{ user_data_folder }}'
+
+      # txt file contains a url refer to remote input data. zip file contains
+      # loacl input data.
+      if ls | grep txt;then
+        txt_file=`ls | grep txt`
+        url=`cat $txt_file`
+        if [ -n "$url" ];then
+          wget $url
+        fi
+      fi
+
+      zip_file=`ls | grep zip`
+
+      unzip $zip_file
+    executable: /bin/bash
+
+- name: Split User Metadata
+  shell: 'cat {{ user_data_folder }}/metadata.yaml'
+  register: user_meta
+
+- name: Get ML metadata yaml content
+  set_fact:
+    user_meta: "{{ user_meta.stdout | from_yaml }}"
+
+- name: Define Parameters - runtime_name
+  shell: echo "{{ user_meta.framework.runtime | regex_findall('[java|go|python|c|c++]') | join("") }}"
+  register: runtime_name
+
+- name: Define Parameters - runtime_version
+  shell: echo "{{ user_meta.framework.runtime | regex_replace('[java|go|python|c|c++]') }}"
+  register: runtime_version
+
+- name: Install {{ user_meta.framework.runtime }} Runtime
+  when: runtime_name
+  shell:
+    cmd: |
+      set -x
+      set -e
+      runtime_name={{ runtime_name.stdout }}
+      runtime_version={{ runtime_version.stdout }}
+      add-apt-repository -y ppa:deadsnakes/ppa
+      apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F60F4B3D7FA2AF80
+      apt update
+      if [ $runtime_name == 'python' ];then
+        apt install -y $runtime_name$runtime_version
+        apt install -y python2.7 python3.5 python3.6 python3.7
+      fi
+
+      if [ $runtime_name == 'java' ];then
+        apt install -y default-jre default-jdk
+        apt install -y openjdk-7-jre openjdk-8-jdk
+      fi
+
+      if [ $runtime_name == 'c' ];then
+        apt install -y gcc=$runtime_version g++=$runtime_version
+        apt install -y gcc-7 g++-7 gcc-8 g++-8 gcc-9 g++-9
+      fi
+
+      if [ $runtime_name == 'go' ];then
+        apt install -y golang-$runtime_version-go
```

Comment:
Again, as I mentioned in previous PS, https://github.com/golang/go/wiki/Ubuntu:

> Note that golang-1.10-go puts binaries in /usr/lib/go-1.10/bin. If you want them on your PATH, you need to make that change yourself.

So, 3 way can make it works:
1. mv the go binary to /usr/bin
2. ln -s ../lib/go-$runtime_version/bin/go /usr/bin/go
3. Add the /usr/lib/go-1.10/bin into your path

Reply:
For now, this demo just supports python runtime.  Other runtime deployments didn't test yet. And the reasons of introducing these conditions is for fufiling the whole framework, let people know that if we want to support another runtimes in the future, we can know the place we can code. ;-). I write your comment into my notes for next works.

Reply:
yeah, Now MOO only support tensorflow+python. Let's enhance it in the future.